test/integration/api/RebalancingAPI.spec.ts

```diff
@@ -1049,24 +1050,84 @@ describe('RebalancingAPI', () => {
           rebalancingSetToken,
           subjectBidQuantity,
-          DEFAULT_CONSTANT_AUCTION_PRICE
+          DEFAULT_CONSTANT_AUCTION_PRICE,
+          expectedTokenAddresses,
         );
 
-        const returnedInflowArray = JSON.stringify(returnedTokenFlowArrays['inflow']);
-        const expectedInflowArray = JSON.stringify(expectedTokenFlowArrays['inflow']);
+        const returnedInflowArray = JSON.stringify(returnedTokenFlowDetailsArrays['inflow']);
+        const expectedInflowArray = JSON.stringify(expectedTokenFlowDetailsArrays['inflow']);
         expect(returnedInflowArray).to.eql(expectedInflowArray);
 
-        const returnedOutflowArray = JSON.stringify(returnedTokenFlowArrays['outflow']);
-        const expectedOutflowArray = JSON.stringify(expectedTokenFlowArrays['outflow']);
+        const returnedOutflowArray = JSON.stringify(returnedTokenFlowDetailsArrays['outflow']);
+        const expectedOutflowArray = JSON.stringify(expectedTokenFlowDetailsArrays['outflow']);
         expect(returnedOutflowArray).to.eql(expectedOutflowArray);
       });
 
+      test('it filters out components with zero units from token flows', async () => {
+        const returnedTokenFlowDetailsArrays = await subject();
+
+        // Get Token Flow bid units not filtered for 0s and count of 0 units
+        const [
+          returnedInflowArray,
+          returnedOutflowArray,
+        ] = await rebalancingSetToken.getBidPrice.callAsync(subjectBidQuantity);
+
+        const returnedInflowZeroCount = returnedInflowArray.reduce((accumulator, unit) => {
+          const bigNumberUnit = new BigNumber(unit);
+          if (bigNumberUnit.eq(0)) {
+            accumulator++;
+          }
+          return accumulator;
+        }, 0);
+
+        const returnedOutflowZeroCount = returnedOutflowArray.reduce((accumulator, unit) => {
+          const bigNumberUnit = new BigNumber(unit);
+          if (bigNumberUnit.eq(0)) {
+            accumulator++;
+          }
+          return accumulator;
+        }, 0);
+
+        // Get Token Flow Details which should filter for 0s and count of 0 units
+        const returnedInflowDetailsZeroCount = returnedTokenFlowDetailsArrays.inflow.reduce((
+          accumulator,
+          component
+          ) => {
+          const bigNumberUnit = new BigNumber(component.unit);
+          if (bigNumberUnit.eq(0)) {
+            accumulator++;
+          }
+          return accumulator;
+        }, 0);
+
+        const returnedOutflowDetailsZeroCount = returnedTokenFlowDetailsArrays.outflow.reduce((
+          accumulator,
+          component
+          ) => {
+          const bigNumberUnit = new BigNumber(component.unit);
+          if (bigNumberUnit.eq(0)) {
+            accumulator++;
+          }
+          return accumulator;
+        }, 0);
+
+        // Ensure there are inflow / outflow components greater than 0
+        expect(returnedInflowZeroCount).to.greaterThan(0);
```

Comment:
Can you be precise about how many there are for these particular bid parameters instead of greater than?